extern/lua/lobject.cpp

```diff
@@ -220,13 +220,15 @@ static lua_Number lua_strx2number (const char *s, char **endptr) {
 #define L_MAXLENNUM	200
 #endif
 
-static const char *l_str2dloc (const char *s, lua_Number *result, int mode) {
-  char *endptr;
-  *result = (mode == 'x') ? lua_strx2number(s, &endptr)  /* try to convert */
-                          : lua_str2number(s, &endptr);
-  if (endptr == s) return NULL;  /* nothing recognized? */
-  while (lisspace(cast_uchar(*endptr))) endptr++;  /* skip trailing spaces */
-  return (*endptr == '\0') ? endptr : NULL;  /* OK if no trailing characters */
+static const char* l_str2dloc(const char* s, lua_Number* result, int mode) {
+  char* endptr;
+  *result = lua_Number(s);
+  if (endptr == s)
+    return NULL; /* nothing recognized? */
```

Comment:
Can I replace the exception with std::optional? This way we can check if the value is "nil" in Lua, if the conversion fails.

Reply:
No, we can't remove the exceptions. They are in a constructor, so there is no way to return an std::optional. We could maybe add a FixedPoint::tryParseFromString() or something similar that could return an optional though. Internally I guess it would just catch the exception. I believe it should already be working, but since we're now relying on it, we should also add a test to make sure FixedPoint doesn't accept trailing characters. Thins function does expect it to accept trailing whitespace though, so that should be handled inside this function as well (I wouldn't add support for that to FixedPoint directly).
Also, we need to return the "endptr". What you are doing atm is undefined behaviour, because you're returning the c_str() from an std::string which will be destroyed when exiting the function. It seems the code that calls this function is only ever checking it against NULL, so maybe an easy fix would be to change the return type to a bool indicating success instead.